From b542ad8784ca135d804f8ae3fd3f09ff21aa203b Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Tue, 11 Jun 2024 17:58:47 +0300
Subject: [PATCH] libtool: Additional flag for test on NetBSD

When linking on NetBSD, the test fails to locate a shared version of
gcc, so a static version of the module is generated, which does not pass
the defined checks. NetBSD seems to be falling back to using dlopen, so
an additional flag has been added for linking.

* tests/flags.at: Add export-dynamic flag on NetBSD.
---
 tests/flags.at | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/tests/flags.at b/tests/flags.at
index fb4109db..e60b649c 100644
--- a/tests/flags.at
+++ b/tests/flags.at
@@ -26,7 +26,17 @@ m4_foreach([lt_tag], [CC, CXX, F77, FC, GCJ],
 AT_KEYWORDS([libtool])
 LT_AT_TAG(m4_defn([lt_tag]))
 
-LDFLAGS="$LDFLAGS -no-undefined"
+# Utilize dlopen to test passing CXX flags through libtool
+case $host_os in
+  netbsd*)
+    if test "CXX" = "lt_tag"; then
+      LDFLAGS="$LDFLAGS -export-dynamic"
+    fi
+    ;;
+  *)
+    LDFLAGS="$LDFLAGS -no-undefined"
+    ;;
+esac
 eval "`$LIBTOOL --config | $EGREP '^(FGREP)='`"
 
 m4_case(lt_tag,$FGREP " $flag_prefix_lib-foo" stdout], [], [ignore])
     else
-- 
2.45.1

